peps/pep-0774.rst

```diff
@@ -0,0 +1,253 @@
+
+.. |targets| replace:: ``Tools/jit/_targets``
+.. _targets: https://github.com/python/cpython/blob/main/Tools/jit/_targets.py
+
+Ignoring the stencils themselves and any necessary JIT README changes, the
+changes to the source code to support reproducible stencil generation and
+hosting are minimal (around 150 lines of changes).
+
+Rejected Ideas
+==============
+
+Several alternative approaches were considered as part of the research and
+exploration for this PEP. However, the ideas below either involve
+infrastructural cost, maintenance burden, or a worse overall developer
+experience.
+
+Using Git submodules
+--------------------
+
+Git submodules are a poor developer experience for hosting stencils because they
+create a different kind of undesirable friction. For instance, any
+updates to the JIT would necessitate regenerating the stencils and committing
+them to a separate repository. This introduces a convoluted process: you must
+update the stencils in the submodule repository, commit those changes, and then
+update the submodule reference in the main CPython repository. This disconnect
+adds unnecessary complexity and overhead, making the process brittle and
+error-prone for contributors and maintainers. 
+
+Using Git subtrees
+------------------
+
+When using subtrees, the embedded repository becomes part of the main
+repository, similar to what’s being proposed in this PEP. However, subtrees
+require additional tooling and steps for maintenance, which adds unnecessary
+complexity to workflows.
+
+Hosting in a separate repository
+--------------------------------
+
+While splitting JIT stencils into a separate repository avoids the storage
+overhead associated with hosting the stencils, it adds complexity to the build
+process. Additional tooling would be required to fetch the stencils and
+potentially create additional and unnecessary failure points in the workflow.
+This separation also makes it harder to ensure consistency between the stencils
+and the CPython source tree, as updates must be coordinated across the
+repositories. Finally, this approach introduces an attack vector, as external
+repositories are less integrated with our workflows, making provenance and
+integrity harder to guarantee.
+
+Hosting in cloud storage
+------------------------
+
+Hosting stencils in cloud storage like S3 buckets or GitHub raw storage
+introduces external dependencies, potentially complicating offline development
+workflows. Also, depending on the provider, this type of hosting comes with
+additional cost, which we’d like to avoid.
+
+Using Git LFS
+--------------
+
+Git LFS adds tool dependency for contributors, complicating the development
+workflow, especially for those who may not already use Git LFS. Git LFS does not
+work well with offline workflows since files managed by LFS require an internet
+connection to fetch when checking out specific commits, which is disruptive for
+even basic Git workflows. 
```

Comment:
You could also mention there's a cost for using LFS (with free quota, but that also complicates things).

https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-git-large-file-storage/about-billing-for-git-large-file-storage